rectools/metrics/auc.py

```diff
@@ -447,22 +479,35 @@ def _get_sufficient_reco_explanation(self) -> str:
             for all users.
             """
 
-    def calc_per_user_from_fitted(self, fitted: AUCFitted) -> pd.Series:
+    def calc_per_user_from_fitted(self, fitted: AUCFitted, is_debiased: bool = False) -> pd.Series:
         """
         Calculate metric values for all users from outer merged recommendations.
 
         Parameters
         ----------
         fitted : AUCFitted
             Meta data that got from `.fit` method.
+        is_debiased : bool, default False
+            If ``True``, indicator that a debias mechanism has been applied before.
+            If ``False``, indicator that the debias mechanism has not been applied before
+            and then it will be applied if it was needed.
 
         Returns
         -------
         pd.Series
             Values of metric (index - user id, values - metric value for every user).
         """
-        outer_merged = fitted.outer_merged_enriched
+        if not is_debiased and self.debias_config is not None:
```

Comment:
**Done**
Since similar functionality is used in several metric classes (auc and classification), I made a generic `_calc_debias` function in `DebiasableMetricAtK.